server/shared/src/main/scala/org/http4s/server/middleware/Throttle.scala

```diff
@@ -62,75 +62,74 @@ object Throttle {
     def local[F[_]](capacity: Int, refillEvery: FiniteDuration)(implicit
         F: Temporal[F]
     ): F[TokenBucket[F]] = {
-      def getTime = F.monotonic.map(_.toNanos)
-      val bucket = getTime.flatMap(time => F.ref((capacity.toDouble, time)))
-
-      bucket.map { counter =>
-        new TokenBucket[F] {
-          override def takeToken: F[TokenAvailability] = {
-            val attemptUpdate = counter.access.flatMap {
-              case ((previousTokens, previousTime), setter) =>
-                getTime.flatMap { currentTime =>
-                  val timeDifference = currentTime - previousTime
-                  val tokensToAdd = timeDifference.toDouble / refillEvery.toNanos.toDouble
-                  val newTokenTotal = Math.min(previousTokens + tokensToAdd, capacity.toDouble)
-
-                  val attemptSet: F[Option[TokenAvailability]] =
-                    if (newTokenTotal >= 1)
-                      setter((newTokenTotal - 1, currentTime))
-                        .map(if (_) Some(TokenAvailable) else None)
-                    else {
-                      val timeToNextToken = refillEvery.toNanos - timeDifference
-                      val successResponse = TokenUnavailable(timeToNextToken.nanos.some)
-                      setter((newTokenTotal, currentTime))
-                        .map(if (_) Some(successResponse) else None)
-                    }
-
-                  attemptSet
+      val getNanoTime = F.monotonic.map(_.toNanos)
+      val refillEveryNanos = refillEvery.toNanos
+      for {
+        // make sure that refillEvery is positive
+        _ <- F.raiseUnless(refillEveryNanos > 0L)(
+          new IllegalArgumentException("refillEvery should be > 0 nano")
```

Comment:
These are repeat of changes in https://github.com/http4s/http4s/pull/6469 right? We should probably merge `series/0.23` into `main` :)

Reply:
Yep, they turned up in here because I had originally based this branch on 0.23

Reply:
Ah, gotcha. I opened https://github.com/http4s/http4s/pull/6499 btw.

Reply:
Apologies, merging in 0.23 created some conflicts here 😅

Reply:
that's fine, I was expecting that :-D